src/coll/mod.rs

```diff
@@ -1196,7 +1202,7 @@ where
 
         let mut cumulative_failure: Option<BulkWriteFailure> = None;
         let mut error_labels: HashSet<String> = Default::default();
-        let mut cumulative_result: Option<InsertManyResult> = None;
+        let mut cumulative_inserted_ids = HashMap::new();
```

Comment:
I initially had just added logic to set the `inserted_ids` on the `cumulative_failure` for failed batches, but I realized that would not correctly handle cases where there are a mix of successful and unsuccessful batches -- for example, if the first batch succeeds and the second batch fails, we would put the `inserted_ids` from batch 1 on the `cumulative_result` but the `inserted_ids` from batch 2 on the `cumulative_failure`, and then only return the failure. 
given this, it seemed simplest to switch to tracking the `inserted_ids` in one place and then just setting them on the `InsertManyResult` / `BulkWriteFailure` at the end.